src/System.Text.Primitives/System/Text/EncodingData.cs

```diff
@@ -365,8 +346,196 @@ private bool VerifySuffix(ReadOnlySpan<byte> buffer, int codeUnitIndex, Symbol s
         // CC = the max value, and DD = the max value's index in the same coord-system as BB.
         struct ParsingTrieNode
         {
-            public byte valueOrNumChildren;
+            public byte ValueOrNumChildren;
             public int IndexOrSymbol;
         }
+
+        private class AbstractParsingTrie
+        {
+            private interface Node
+            {
+            }
+
+            private struct InternalNode : Node
```

Comment:
Since you are boxing this struct anyway to cast it to the interface, it would be better to either make it a class that inherits from List<Node>; or to have just one struct type with SymbolByteOrIndex field.